kernel/aster-nix/src/process/clone.rs

```diff
@@ -311,6 +313,10 @@ fn clone_child_process(
         clone_flags,
     )?;
 
+    // clone namespaces and switch to new namespaces
+    let child_namespaces = clone_namespaces(&child, current.namespaces(), clone_flags);
+    child.switch_namespaces(child_namespaces);
```

Comment:
I think this part of the code should be placed in the initialization code of the child. After the child is created, it should follow the `clone_flags` and create the namespace accordingly.

Is there any difficulty in the implementation here? The API design appears to always execute `switch_namespace`, giving the impression that the namespace will be switched regardless of whether `CLONE_NEWNS` is present or not.